src/SoftApService.js

```diff
@@ -59,17 +61,28 @@ class SoftAPService {
       idx: index,
       pwd: encryptedPassword,
       sec: security,
-      'ssid-value': ssid,
+      ssid,
     });
 
-    await fetchJSON(`${BASE_URL}/configure-ap`, {
+    const { r: responseCode } = await fetchJSON(`${BASE_URL}/configure-ap`, {
       body,
       headers: {
         ...HEADERS,
         'Content-Length': body.length,
       },
       method: 'POST',
     });
+
+    if (responseCode === INCORRECT_WIFI_PASSWORD_CODE) {
+      throw new Error('Incorrect Wifi password!');
+    }
+
+    if (responseCode !== SUCCESS_RESPONSE_CODE) {
```

Comment:
it sometimes throws with -2 code..idk why its happening actually. and if I send the same credentials again and again(click send button after the error) it goes successfully on 3-4 try, so its is probably not absolutely critical if we won't be able to fix it.